src/tir/usmp/algo/greedy.cc

```diff
@@ -39,118 +39,14 @@
 #include <tvm/tir/builtin.h>
 #include <tvm/tir/function.h>
 #include <tvm/tir/stmt_functor.h>
+#include <tvm/tir/usmp/algo/greedy.h>
 #include <tvm/tir/usmp/utils.h>
 
 namespace tvm {
 namespace tir {
 namespace usmp {
 namespace algo {
 
-/*!
- * \brief This is the base class for Greedy Algorithms where the sorting
- * is specialized in the extended classes based on the greedy criteria.
- */
-class GreedyBase {
```

Comment:
I think we should only move the declaration of the class to the header and we should keep the definition here.

Reply:
Ok. Moved implementation to greedy.cc